eitprocessing/datahandling/intervaldata.py

```diff
@@ -0,0 +1,185 @@
+import copy
+import itertools
+from dataclasses import dataclass, field
+from typing import Any, NamedTuple, TypeVar
+
+import numpy as np
+from typing_extensions import Self
+
+from eitprocessing.datahandling.mixins.equality import Equivalence
+from eitprocessing.datahandling.mixins.slicing import HasTimeIndexer
+
+T = TypeVar("T", bound="IntervalData")
+
+
+class TimeRange(NamedTuple):
+    """A tuple containing the start time and end time of a time range."""
+
+    start_time: float
+    end_time: float
```

Comment:
NamedTuples are like dataclasses, but even lighter. The reason to use them over a normal tuple is that you can address the items using either the name or a integer getter:

```python
tr = TimeRange(10, 20)
tr2 = TimeRange(end_time=20, start_time=10)

assert tr == tr2
assert tr[0] is tr.start_time
```

Good call on nomenclature. I'll rename to Interval.

Reply:
Apparently it is not possible to subclass a slice object. But we can work around this by adapting `__getitem__` for all time-sliceable objects. E.g.:

```python
class Something:
    def __getitem__(self, key):
        if isinstance(key, TimeRange):
            # use TimeRange as slice
```

Reply:
I created #203 to handle this.

Reply:
Have you given this any thought?

> I do think we could give it some additional features though ((it may need to become a dataclass at that point, not sure how far the functionality of named tuple goes).
> I am thinking e.g. of some checks like that `start_time` is smaller than `end_time`, or no negative numbers are given (I am correct in thinking that that shouldn't be possible right?).
> Even then, I'm not positive what the added benefit of a separate class is over an internal checking function.

Reply:
It makes sense to add some initialisation checks, which would more easily be implemented in a dataclass using `__post_init__`. 

It makes sense to do these checks at initialisation to remove that burden from any class using TimeRanges.

I created #207.